analysis/tables_ONS_TPP_reg_dereg.R

```diff
@@ -0,0 +1,151 @@
+# Preliminaries ----
+
+# Import libraries
+library("tidyverse")
+library("dtplyr")
+library("lubridate")
+library("glue")
+library("here")
+
+
+## Create output directory
+output_dir <- here("output", "report", "reg_dereg_ONS_TPP")
+fs::dir_create(output_dir)
+
+# Import processed data ----
+dataset0 <- read_csv("output/dataset_death_date_diff.csv.gz") %>% 
+  mutate(
+    TPP_death_date = as.Date(TPP_death_date),
+    ons_death_date = as.Date(ons_death_date),
+    date_of_birth = as.Date(date_of_birth),
+    age_band = as.factor(age_band),
+    practice = as.factor(practice),
+    ons_death_place = as.factor(ons_death_place),
+    region = as.factor(region),
+    IMD_q10 = as.factor(IMD_q10),
+    ethnicity = as.factor(ethnicity),
+    sex = as.factor(sex),
+    last_registration_start_date = as.Date(last_registration_start_date),
+    last_registration_end_date = as.Date(last_registration_end_date)
+  )
+
+# Rounding function
+
+rounding <- function(vars) {
+  case_when(vars == 0 ~ 0,
+            vars > 7 ~ round(vars / 5) * 5)
+}
+
+# Data curating ---------------
+
+dataset <- dataset0 %>%
+  mutate(
+    TPP_death_reg = case_when(
+      !is.na(TPP_death_date) & !is.na(last_registration_start_date) ~
+        as.Date(TPP_death_date) - as.Date(last_registration_start_date),
+      TRUE ~ as.difftime(NA_real_, units = "days")
+    ),
+    TPP_death_dereg = case_when(
```

Comment:
there should be an option for if the last registration end date is missing - we should quantify how commonly that happens